main.cpp

```diff
@@ -18,16 +18,45 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter_;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) { }
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+        const auto *Item = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        if (Item ->getCastKind() == CK_ToVoid)
+            return;
+        if (Item ->getExprLoc().isMacroID())
+            return;
+        SourceManager &SourceM = *Result.SourceManager;
+
+        auto ReRange = CharSourceRange::getCharRange (Item->getLParenLoc(),
+                                                      Item->getSubExprAsWritten()->getBeginLoc());
+
+        StringRef DestTypeString =
+                Lexer::getSourceText(CharSourceRange::getTokenRange(Item->getLParenLoc().getLocWithOffset(1),
+                                                                    Item->getRParenLoc().getLocWithOffset(-1)),
+                                                                    SourceM,
+                                                                    Result.Context->getLangOpts());
+
+        std::string str = ("static_cast<" + DestTypeString + ">").str();
+        const Expr *SubExpr = Item->getSubExprAsWritten()->IgnoreImpCasts();

```

Comment:
Please describe why you are using here `getSubExprAsWritten()` and `IgnoreImpCasts()`.
What would be the difference with the use of `getSubExprAsWritten()->IgnoreImpCasts()`, `getSubExpr()->IgnoreImpCasts()` and `getSubExprAsWritten()`?

Reply:
There is no answer to the question.